concent_api/core/exceptions.py

```diff
@@ -1,11 +1,19 @@
+from utils.constants import ErrorCode
+
 
 class UnexpectedResponse(Exception):
     pass
 
 
-class Http400(Exception):
+class ConcentInSoftShutdownMode(Exception):
     pass
 
 
-class ConcentInSoftShutdownMode(Exception):
-    pass
+class Http400(Exception):
+
+    def __init__(self, error_message: str, error_code: ErrorCode, *args, **kwargs) -> None:
+        assert isinstance(error_message, (str, None))
+        assert isinstance(error_code, ErrorCode)
+        self.error_code = error_code
+        self.error_message = '' if error_message is None else error_message
+        super().__init__(error_message, args, kwargs)
```

Comment:
what do we need `args` and `kwargs` for?

Reply:
`Exception` class accepts it in its `__init__` method, so the class which inherits from it should accept it too.

Reply:
Why? You can call `super().__init__(error_message)` ;) You do not make any use of `args` and `kwargs` in your class and you don't expect anyone to `raise Http400` with any other arguments than `error_code` and `error_message`. 
For me having `args` and `kwargs` make thing unnecessarily confusing.
(https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions)

Reply:
I know, but if the upper class `__init__` accepts `args` and `kwargs`, so I think that inheriting class and overriding a particular method should keep signature as close to the overridden a possible, just in case someone would try to use it. It is rather a matter of good practice, not this particular use case.

Reply:
@rwrzesien, in staticaly typed languages like Java or C++ I wouldn't argue with that. However, in python it adds no value, because you do not have strict polymorphism, i. e., everything is checked at run-time (because there is no compile-time). 
I would really like not to have `args` and `kwargs` here, but I leave up to you ;)

Reply:
I think it's fine to have `args` and `kwargs` in case someone wants to add extra info. `BaseException.__init__()` stores anything passed to it on `self` so I see how someone might find it useful (though I've rarely had an actual use for it). We don't have to break this feature, though I share the sentiment that the extra args make the code uglier :)

But if you're passing the arguments down to the inherited init, isn't this superfluous?
``` python
        self.error_code = error_code
        self.error_message = '' if error_message is None else error_message
```

Reply:
> But if you're passing the arguments down to the inherited init, isn't this superfluous?

The point is to allow inheriting class to use features (by passing arguments) to parent class method. But in this case there are no such features, so I have done it rather automatically, as a good practice. You are both right that in this case this is not necessary, so I have removed it.

Reply:
> But in this case there are no such features

There are. I said it my comment. `BaseException` stores the parameters you give it on `self`. So you don't have to do it yourself.